src/components/Api.js

```diff
@@ -0,0 +1,204 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    // baseUrl: "https://around-api.en.tripleten-services.com/v1"
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  /*---------------------------------------------------*/
+  /*         Profile Picture Api functions             */
+  /*---------------------------------------------------*/
+
+  // – Update avatar
+  // PATCH https://around-api.en.tripleten-services.com/v1/users/me/avatar
+  patchProfileAvatar(link) {
+    return fetch(this._baseUrl + "/users/me/avatar", {
+      method: "PATCH",
+      headers: {
+        authorization: this._headers.authorization,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        avatar: link,
+      }),
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status} + ${res.message}`);
+      })
+      .then((result) => {
+        return result;
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.